tests/Select.spec.js

```diff
@@ -5,10 +5,12 @@ import KeyCode from 'rc-util/lib/KeyCode';
 import Select, { Option, OptGroup } from '../src';
 import focusTest from './shared/focusTest';
 import blurTest from './shared/blurTest';
+import keyDownTest from './shared/keyDownTest';
 
 describe('Select', () => {
   focusTest('single');
   blurTest('single');
+  keyDownTest('single');
```

Comment:
@yesmeck should I change to `combobox` ? Test passes both cases.

Reply:
You can add `combobox` to here https://github.com/react-component/select/blob/master/tests/Select.combobox.spec.js#L10

Reply:
Ok.. so shall I remove it from `Select.spec.js` ?

Reply:
I'd prefer keeping it.

Reply:
`single` mode need enable `showSearch` to show input.

Reply:
`showSearch` is enabled in the spec ?

Reply:
@yesmeck Did you mean that the `onInputKeyDown` prop should only fire when `showSearch` is enabled for `single` mode ?

Reply:
Yes. Because there is not input in `single` mode if `showSearch` is `false`.

Reply:
Ok.. so a simple check inside the `chaining` function is sufficient I guess. Any other edge cases you can think of before I commit ?

Reply:
Never mind, I forgot `showSearch` is default to `true` in `rc-select`.